src/services/packet_processor.rs

```diff
@@ -37,6 +38,15 @@ pub fn start_inbound<
 
                 let packet = read(&mut msg.cursor.clone(), translation_data.state);
                 let packet = translate(packet, translation_data.clone());
+
+                // Send raw packet info if we provided a channel
+                let test_sender_clone = test_sender.clone();
+                if let Some(test_sender_clone) = test_sender_clone {
+                    test_sender_clone
+                        .send((translation_data.state, packet.clone()))
+                        .expect("Failed to send packet to channel");
+                }
```

Comment:
What are we trying to accomplish with this? Are we trying to test our ability to read packets from raw data into structs?

Reply:
My thought process was that to avoid lots of code duplication, we need a way to send data to the test thread to see the data. We cannot access the data that's inside the packets - it's what we are trying to see if its contents are what we expected - in the functions

Reply:
Why don't we just use the receiver this already has?